packages/vscode/src/plugins/prisma-postgres-manager/PrismaPostgresRepository.ts

```diff
@@ -142,14 +152,37 @@ export class PrismaPostgresApiRepository implements PrismaPostgresRepository {
   triggerRefresh() {
     void this.credentialsStore.reloadCredentialsFromDisk()
     // Wipe caches
+    this.regionsCache = []
     this.projectsCache = new Map()
     this.remoteDatabasesCache = new Map()
     // Trigger full tree view refresh which will fetch fresh data on demand due to cache misses
     this.refreshEventEmitter.fire()
   }
 
-  getRegions(): Promise<string[]> {
-    return Promise.resolve(regions.slice())
+  async getRegions(): Promise<Region[]> {
+    if (this.regionsCache.length > 0) return this.regionsCache
+
+    const workspaceId = (await this.getWorkspaces()).at(0)?.id
+    if (!workspaceId) return Promise.resolve([])
```

Comment:
```suggestion
    if (!workspaceId) return []
```
it's already an `async` function